File: zz_good-bye/hyeonguk/2023_7/20230722/BOJ_25206_너의평점은.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		double result = 0;
		double totalScore = 0;
		Scanner sc = new Scanner(System.in);
		
		for(int tc=0; tc<20; tc++) {
			String subject = sc.next();
			double score = sc.nextDouble();
			String grade = sc.next();
			
			totalScore += score;
			if(grade.equals("P")) {
				totalScore -= score;
			}
			
			if(grade.equals("A+")) {
				result += 4.5*score;
			}else if(grade.equals("A0")) {
				result += 4.0*score;
			}else if(grade.equals("B+")) {
				result += 3.5*score;
			}else if(grade.equals("B0")) {
				result += 3.0*score;
			}else if(grade.equals("C+")) {
				result += 2.5*score;
			}else if(grade.equals("C0")) {
				result += 2.0*score;
			}else if(grade.equals("D+")) {
				result += 1.5*score;
			}else if(grade.equals("D0")) {
				result += 1.0*score;
			}
		}
		System.out.printf("%f", result/totalScore);
		sc.close();
	}
}

File: woojae/2024_5/0518/BOJ_25206_너의 평점은.py
info = {'A+': 4.5,
        'A0': 4.0,
        'B+': 3.5,
        'B0': 3.0,
        'C+': 2.5,
        'C0': 2.0,
        'D+': 1.5,
        'D0': 1.0,
        'F': 0.0}

answer = 0
total_score = 0

for _ in range(20):
    data = input()
    subject, score, grade = data.split()

    if grade == 'P':
        continue

    answer += float(score) * float(info[grade])
    total_score += float(score)
    
answer /= total_score

print(answer)
